test/client_test.coffee

```diff
@@ -143,6 +146,35 @@ describe 'Basic test for the 3Scale::Client', ->
       nock.cleanAll()
 
 
+  describe 'The oauth_authrep method', ->
+    it 'should throw an exception if is called without :app_id', ->
+      client = new Client()
+      assert.throws (() ->  client.authrep_with_oauth({}, () ->)), 'missing app_id'
+
+
+  describe 'Request headers in authrep_with_oauth calls', ->
+    it 'should throw an exception if authrep_with_oauth called without :app_id', ->
+      client = new Client()
+      assert.throws (() -> client.authrep_with_oauth({}, () ->)), 'missing app_id'
+
+    it 'should include the Host and X-3scale-User-Agent headers', (done) ->
+      opts =
+        reqheaders:
+          'Host': 'su1.3scale.net'
+          'X-3scale-User-Agent': 'plugin-node-v' + require('../package.json').version
```

Comment:
I think all OAuth calls should be sending the header `X-3scale-OAuth2-Grant-Type` but maybe @mpguerra can comment if that is critical or not as we are not specifically implementing one flow or another here.